iOS Client/PodcastsController/PodcastDetailViewController.swift

```diff
@@ -79,17 +117,27 @@ class PodcastDetailViewController: UIViewController, StoreSubscriber {
     
     func newState(state: MBAppState) {
         switch state.podcastsState.player {
-        case .error, .initialized, .paused, .finished:
-            playPauseButton.setImage(#imageLiteral(resourceName: "play"), for: .normal)
+        case .initialized, .error, .paused, .finished:
+            playPauseButton.setImage(#imageLiteral(resourceName: "play-arrow"), for: .normal)
+            if playerState == .playing, playerState != .initialized { animateImage(size: CGSize(width: 150, height: 150), duration: 0.6)}
         case .playing:
-            playPauseButton.setImage(#imageLiteral(resourceName: "pause"), for: .normal)
+            playPauseButton.setImage(#imageLiteral(resourceName: "pause-bars"), for: .normal)
+            if playerState != .playing, playerState != .initialized { animateImage(size: CGSize(width: 200, height: 200), duration: 0.6)}
         }
+        playerState = state.podcastsState.player
         titleLabel.text = state.podcastsState.selectedPodcast?.title
+        if let imageName = state.podcastsState.selectedPodcast?.image {
+           imageView.image = UIImage(named: imageName)
+            backgroundImageView.image = UIImage(named: imageName)
+        }
+        self.navigationItem.title = state.podcastsState.selectedPodcast?.podTitle
     }
     
     func updateCurrentDuration(current: Double, total: Double ) {
         totalDuration = total
-        durationLabel.text = "\(formatter.string(from: NSNumber(value: current)) ?? "0") / \(formatter.string(from: NSNumber(value: total)) ?? "0")"
+        timeFormatter?.allowedUnits = total >= Double(3600) ? [.hour, .minute, .second] : [.minute, .second]
```

Comment:
nice, so, if this is derived each time, can we delete line 67 in the configureFormatter function?